src/hotspot/share/gc/g1/g1EvacFailureRegions.hpp

```diff
@@ -55,8 +61,19 @@ class G1EvacFailureRegions {
 
   bool contains(uint region_idx) const;
   void par_iterate(HeapRegionClosure* closure,
-                   HeapRegionClaimer* _hrclaimer,
+                   HeapRegionClaimer* hrclaimer,
                    uint worker_id) const;
+  void initialize_chunks(uint active_workers, const char* task_name);
+  // Iterate through all chunks in regions that failed evacuation during the entire collection.
+  void par_iterate_chunks_in_regions(G1HeapRegionChunkClosure* chunk_closure,
+                                     uint worker_id) const;
+
+  // Do necessary preparation for evacuation failure regions
+  void prepare_regions();
+
+  uint max_regions() const {
+    return _max_regions;
+  }
```

Comment:
Looking at how this field is used:

1. `region_idx < _max_regions`; can be caught the underlying bitmap
2.  `_max_regions = 0; // To have any record() attempt fail...`; can be replaced by a null-check of `_evac_failure_regions`.

This issue is preexisting though.